File: hw/calle/u1/hw11ChikenCoops/ChickenCoopsJavaScript/ChickenCoop.js
const Chicken = require("./Chicken");

class ChickenCoop {
    constructor(id, name) {
        this._id = id;
        this._name = name;
        this._chickens = []; 
    }

    get_id() {
        return this._id;
    }

    get_name() {
        return this._name;
    }

    get_chickens() {
        return this._chickens;
    }

    set_id(id) {
        this._id = id;
    }

    set_name(name) {
        this._name = name;
    }

    addChicken(chicken) {
        this._chickens.push(chicken);
    }

    removeChicken(chickenId) {
        this._chickens = this._chickens.filter(chicken => chicken.get_id() !== chickenId);
    }

    getChickenCount() {
        return this._chickens.length;
    }

    toString() {
        let coopInfo = `Chicken Coop ID: ${this._id}\n` +
                      `Name: ${this._name}\n` +
                      `Number of Chickens: ${this.getChickenCount()}\n`;
        
        if (this._chickens.length > 0) {
            coopInfo += "Chickens in this coop:\n";
            this._chickens.forEach(chicken => {
                coopInfo += `- ${chicken.get_name()} (ID: ${chicken.get_id()})\n`;
            });
        } else {
            coopInfo += "No chickens in this coop.\n";
        }
        
        return coopInfo;
    }

    displayAllChickens() {
        console.log(`\n=== Detailed info for ${this._name} ===`);
        this._chickens.forEach(chicken => {
            console.log(chicken.toString());
            console.log("---");
        });
    }
}

module.exports = ChickenCoop;

File: hw/calle/u1/hw11ChikenCoops/ChickenCoopsPython/ChickenCoop.py
class ChickenCoop:
    def __init__(self, id, name):
        self._id = id
        self._name = name
        self._chickens = [] 

    def get_id(self):
        return self._id

    def get_name(self):
        return self._name

    def get_chickens(self):
        return self._chickens

    def set_id(self, id):
        self._id = id

    def set_name(self, name):
        self._name = name

    def add_chicken(self, chicken):
        self._chickens.append(chicken)

    def remove_chicken(self, chicken_id):
        self._chickens = [chicken for chicken in self._chickens if chicken.get_id() != chicken_id]

    def get_chicken_count(self):
        return len(self._chickens)

    def __str__(self):
        coop_info = (f"Chicken Coop ID: {self._id}\n"
                    f"Name: {self._name}\n"
                    f"Number of Chickens: {self.get_chicken_count()}\n")
        
        if self._chickens:
            coop_info += "Chickens in this coop:\n"
            for chicken in self._chickens:
                coop_info += f"- {chicken.get_name()} (ID: {chicken.get_id()})\n"
        else:
            coop_info += "No chickens in this coop.\n"
        
        return coop_info

    def display_all_chickens(self):
        print(f"\n=== Detailed info for {self._name} ===")
        for chicken in self._chickens:
            print(chicken)
            print("---")